Tests/APITesters/RevenueCatUIAPITester/SwiftAPITester/PaywallViewControllerAPI.swift

```diff
@@ -19,6 +19,16 @@ func paywallViewControllerAPI(_ delegate: Delegate, _ offering: Offering?) {
     let _: UIViewController = PaywallViewController(offering: offering, displayCloseButton: true)
 }
 
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, *)
+func paywallFooterViewControllerAPI(_ delegate: Delegate, _ offering: Offering?) {
+    let controller = PaywallFooterViewController()
+    controller.delegate = delegate
+
+    let _: UIViewController = PaywallFooterViewController(offering: offering)
+    let _: UIViewController = PaywallFooterViewController(displayCloseButton: true)
```

Comment:
One of the consequences of inheriting from `PaywallViewController` is that we get this constructor, which shouldn't exist in the footer version...

Reply:
You can override it and mark it unavailable:
```
@available(*, unavailable)
override init(displayCloseButton: Bool){}
```